app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,64 @@
+class RentalsController < ApplicationController
+
+  def overdue
+  end
+
+  def checkout
+    rental = Rental.new(rental_params)
+
+    # if rental.movie.available_inventory
+    if rental.save
+      render(
+        json: {
+          customer_id: rental.customer_id,
+          movie_id: rental.movie_id,
+          due_date: rental.due_date
+        },
+        status: :ok
+      )
+      rental.customer.movies_checked_out_count +=1
+      rental.movie.available_inventory -= 1
```

Comment:
In order to update the values in the DB, you'll have to call `save` on `rental.customer` and `rental.movie`. If you're looking for some additional reading, this sort of situation (where several models need to be updated all at once) [is a good candidate for wrapping in a database transaction](http://markdaggett.com/blog/2011/12/01/transactions-in-rails/).

This sort of logic would be great as a model method - perhaps a model filter (similar to a controller filter, but happens before writing to the DB).

Reply:
The other option here would be to not track these as separate columns in the DB, but generate them dynamically whenever they're needed by counting rentals associated with this movie/customer.